src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -239,24 +243,89 @@ public bool TryRebaselineCurrentSarifError(string uriBaseId, string originalFile
 
             string rebaselinedFile;
 
-            if (Uri.IsWellFormedUriString(originalFilename, UriKind.Absolute))
+            if (FileDetails.ContainsKey(originalFilename))
             {
-                rebaselinedFile = DownloadFile(originalFilename);
+                // File contents embedded in SARIF.
+                rebaselinedFile = CreateFileFromContents(originalFilename);
             }
             else
             {
-                rebaselinedFile = GetRebaselinedFileName(uriBaseId, originalFilename);
-            }
+                if (Uri.IsWellFormedUriString(originalFilename, UriKind.Absolute))
+                {
+                    // File needs to be downloaded.
+                    rebaselinedFile = DownloadFile(originalFilename);
+                }
+                else
+                {
+                    // User needs to locate file.
+                    rebaselinedFile = GetRebaselinedFileName(uriBaseId, originalFilename);
+                }
 
-            if (String.IsNullOrEmpty(rebaselinedFile) || originalFilename.Equals(rebaselinedFile, StringComparison.OrdinalIgnoreCase))
-            {
-                return false;
+                if (String.IsNullOrEmpty(rebaselinedFile) || originalFilename.Equals(rebaselinedFile, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
             }
 
             CurrentSarifError.RemapFilePath(originalFilename, rebaselinedFile);
             return true;
         }
 
+        // Contents are embedded in SARIF. Create a file from these contents.
+        private string CreateFileFromContents(string fileName)
+        {
+            var fileData = FileDetails[fileName];
+
+            // Get temporary path.
+            string path = Path.GetTempPath();
```

Comment:
It's confusing that `path` means two different things:
1. The path to the temporary directory.
2. The full path to the stored file.

I suggest:
1. Declare an instance variable `temporaryFileDirectoryPath`. Initialize it in the ctor, using the logic shown here.
2. Then declare `path` here (not initialized yet).
3. Then:
```
if (fileName.StartsWith(temporaryFileDirectoryPath))
{
    path = fileName;
}
{
    ...
}
```

Reply:
As we discussed: The improved name `finalPath` makes the code clear.

---
In reply to: [137895686](https://github.com/Microsoft/sarif-sdk/pull/618#discussion_r137895686) [](ancestors = 137895686)